blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/BlueTestResultFactory.java

```diff
@@ -0,0 +1,90 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+import io.jenkins.blueocean.rest.Reachable;
+import io.jenkins.blueocean.rest.model.BlueTestResult;
+import io.jenkins.blueocean.rest.model.BlueTestSummary;
+import jenkins.model.Jenkins;
+
+public abstract class BlueTestResultFactory implements ExtensionPoint {
```

Comment:
Hmm, I remember going over with @kohsuke and for plugin to serve tests they must implement `AbstractTestResultAction`. e.g. [robot-plugin](https://github.com/jenkinsci/robot-plugin/blob/master/src/main/java/hudson/plugins/robot/RobotBuildAction.java) or [test-ng plugin](https://github.com/jenkinsci/testng-plugin-plugin/blob/25ff107f8cab86a4963bfafa987132130431d8e0/src/main/java/hudson/plugins/testng/TestNGTestResultBuildAction.java). Thats how they even appear in the classic UI. 

Its understood, rest of the path will be specific to a test plugin but then they will implement UI test extension to serve their views where they can call their action specific path. I guess where you are going with this is that, they can implement this extension point and their test results will start appearing in blueocean, which is great. this is going to be easier, I think, for existing plugin developer to simply implement backend extension. 👍